documentation/src/main/asciidoc/topics/con_performance_persistence.adoc

```diff
@@ -0,0 +1,39 @@
+[id='performance-persistence_{context}']
+= Persistence
+
+Configuring {brandname} to interact with a persistent data source has the most significant performance impact of any other capability.
+This performance penalty comes from the fact that more traditional data sources are inherently slower than in-memory caches.
+Read and write operations will always take longer when the call goes to non-volatile memory outside the JVM.
+
+Depending on how you use cache stores, though, the reduction of {brandname} performance is offset by the gains that come from being able to access data in memory.
+Persistent storage might slow down {brandname} but {brandname} still gives you a much faster data tier for your applications.
+Using persistent storage also gives other benefits such as allowing you to preserve state for graceful cluster shutdowns.
+You can also overflow data from your caches to persistent storage and gain capacity beyond what is available in memory only.
+For example, you can have ten million entries in total while keeping only two million of them in memory.
+
+.Writing modes
+
+{brandname} adds key/value pairs to caches and persistent storage in either write-through mode or write-behind mode.
+These writing modes have different impacts on performance so it is important to take them into account when planning a {brandname} deployment.
+
+* Write-through caching
++
+{brandname} writes data to the cache and persistent storage simultaneously, which increases consistency but adds latency to write operations.
+`Cache.put()` calls result in application threads waiting until writes to persistent storage complete.
+
+* Write-behind caching
++
+{brandname} synchronously writes data to the cache but then adds the modification to a queue so that the write to persistent storage happens asynchronously, which decreases consistency but reduces latency of write operations.
+However, there is a limit to how much performance write-behind can provide.
+If too many write operations enter the queue, {brandname} cannot process them fast enough to clear the queue.
+// Review comment
+// Will: What can we say here? What problems result from too many enqueued writes?
+
+.Pre-loading
+
+Another aspect of persistent storage that can adversely affect {brandname} cluster performance is pre-loading caches.
+This capability populates your caches with frequently accessed data when {brandname} clusters start so they are "warm" and can handle reads and writes straight away.
+You do need to be cautious about pre-loading too many entries because that can cause {brandname} cluster to have much slower start up times.
```

Comment:
That sounds like a misconfiguration to me @wfink. You mean if the user configures eviction so that Infinispan nodes cannot hold all the pre-loaded data then the pre-load is a waste of time?

Reply:
if eviction is configured, it preloads only a partial number of keys.